### PR TITLE
Publish request for kel-agent

### PR DESCRIPTION
**Pull request by xylo04**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link: https://github.com/k0swe/kel-agent/graphs/contributors**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: radio.k0swe.kel_agent.Kel_Agent.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+app-id: radio.k0swe.kel_agent.Kel_Agent
```
Comment (hfiguiere) [inline radio.k0swe.kel_agent.Kel_Agent.yml:1]: This app-id doesn't follow the rules as the domain doesn't exist.

**Comment by hfiguiere** [comment]
bot, build  radio.k0swe.kel_agent.Kel_Agent

File: radio.k0swe.kel_agent.Kel_Agent.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        tag: v2.1.0
+        commit: 0134bfd19e3e9eb1930ca6b86d07d063cf77ec74
+        dest: vendor/github.com/k0swe/wsjtx-go/v2/
+      - type: git
+        url: https://github.com/leemcloughlin/jdn.git
+        commit: 6f88db6a6bf2176674100d1a1c692596fcf6e2ba
+        dest: vendor/github.com/leemcloughlin/jdn/
+
+      - type: git
+        url: ssh://git@github.com/k0swe/kel-agent.git
```
Comment (hfiguiere) [inline radio.k0swe.kel_agent.Kel_Agent.yml:33]: you should use the https URL here.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
bot, build radio.k0swe.kel_agent.Kel_Agent

File: radio.k0swe.kel_agent.Kel_Agent.yml (hunk 1 above)
Comment (xylo04) [reply radio.k0swe.kel_agent.Kel_Agent.yml]: Changed to radio.k0swe.Kel_Agent. I do control https://www.k0swe.radio/.

File: radio.k0swe.kel_agent.Kel_Agent.yml (hunk 2 above)
Comment (xylo04) [reply radio.k0swe.kel_agent.Kel_Agent.yml]: Done.

**Comment by hfiguiere** [comment]
bot, build radio.k0swe.Kel_Agent

File: radio.k0swe.Kel_Agent.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+command: kel-agent
+finish-args:
+  - --share=network
+modules:
+  - name: kel-agent
+    sources:
+      - type: git
+        url: https://github.com/gorilla/websocket.git
+        tag: v1.4.2
+        commit: e8629af678b7fe13f35dff5e197de93b4148a909
```
Comment (hfiguiere) [inline radio.k0swe.Kel_Agent.yml]: tag and commit are mismatched.

File: radio.k0swe.Kel_Agent.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        dest: vendor/github.com/gorilla/websocket/
+      - type: git
+        url: https://github.com/k0swe/wsjtx-go.git
+        tag: v2.1.0
+        commit: 0134bfd19e3e9eb1930ca6b86d07d063cf77ec74
```
Comment (hfiguiere) [inline radio.k0swe.Kel_Agent.yml]: same here, tag and commit are mistmatched.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: radio.k0swe.Kel_Agent.yml (hunk 3 above)
Comment (xylo04) [reply radio.k0swe.Kel_Agent.yml]: Woops, fixed in https://github.com/flathub/flathub/pull/2360/commits/394522cc87910f04feaabfc21455f5602b4ad520

File: radio.k0swe.Kel_Agent.yml (hunk 4 above)
Comment (xylo04) [reply radio.k0swe.Kel_Agent.yml]: https://github.com/flathub/flathub/pull/2360/commits/394522cc87910f04feaabfc21455f5602b4ad520

File: radio.k0swe.Kel_Agent.yml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+runtime: org.freedesktop.Platform
+runtime-version: '20.08'
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.golang
+build-options:
+  append-path: /usr/lib/sdk/golang/bin
+  env:
+    - 'LDFLAGS='
+command: kel-agent
```
Comment (hfiguiere) [inline radio.k0swe.Kel_Agent.yml:11]:
```suggestion
command: kel-agent
rename-desktop-file: radio.k0swe.kel_agent.Kel_Agent.desktop
rename-appdata-file: radio.k0swe.kel_agent.Kel_Agent.metainfo.xml
rename-icon: radio.k0swe.kel_agent.Kel_Agent
```

This is needed to match.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
bot, build radio.k0swe.Kel_Agent

**Comment by xylo04** [comment]
I'm making an upstream change to make the app ID consistent

**Comment by hfiguiere** [comment]
> I'm making an upstream change to make the app ID consistent

Yeah that's better. I wrongly assumed that `rename-appdata-file` did fix the app-id too. It doesn't.

**Comment by hfiguiere** [comment]
bot, build radio.k0swe.Kel_Agent

**Comment by xylo04** [comment]
```
builddir/files/share/appdata/radio.k0swe.Kel_Agent.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• style-invalid         : Not enough <screenshot> tags, minimum is 1
• tag-missing           : <release> required
```

Looks like I need to fix that up, upstream again.

**Comment by hfiguiere** [comment]
bot, build radio.k0swe.Kel_Agent

File: radio.k0swe.Kel_Agent.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        dest: vendor/github.com/leemcloughlin/jdn/
+
+      - type: git
+        url: https://github.com/k0swe/kel-agent.git
+        tag: v0.3.5
+        commit: edcd7dfd4e60153c1972e148f8912478022407d8
+    buildsystem: simple
+    build-commands:
+      - cp assets/modules.txt vendor/
+      - make
```
Comment (barthalion) [inline radio.k0swe.Kel_Agent.yml:39]: It does not particularly matter but you can parallelize it with `-j$FLATPAK_BUILDER_N_JOBS`

File: radio.k0swe.Kel_Agent.yml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+      - install -D kel-agent /app/bin/kel-agent
+      - install -Dm644 assets/radio.k0swe.Kel_Agent.svg /app/share/icons/hicolor/scalable/apps/radio.k0swe.Kel_Agent.svg
+      - install -Dm644 assets/radio.k0swe.Kel_Agent.metainfo.xml /app/share/metainfo/radio.k0swe.Kel_Agent.metainfo.xml
+      - install -Dm644 assets/radio.k0swe.Kel_Agent.desktop /app/share/applications/radio.k0swe.Kel_Agent.desktop
+    cleanup:
+      - /lib/debug
```
Comment (barthalion) [inline radio.k0swe.Kel_Agent.yml:45]:
```suggestion
```

Debug symbols are automatically split to separate extension and are not installed by default.

File: radio.k0swe.Kel_Agent.yml (hunk 6 above)
Comment (xylo04) [reply radio.k0swe.Kel_Agent.yml]: Done

**Comment by AsciiWolf** [comment]
@xylo04 Thanks for working on ham radio related Flatpak! Feel free to let me know if you are interested in co-maintaining other two ham radio apps on Flathub: https://github.com/flathub/com.w1hkj.fldigi and https://github.com/flathub/com.w1hkj.flrig

The original maintainer left and I am the only maintainer of these apps.

There were also some other [ham app Flatpaks](https://github.com/flathub/flathub/issues?q=author%3Alu9dce) made by this maintainer, most of the PRs were ready, but then got closed by their author before being actually published on Flathub. It would be great to continue work on these Flatpaks. :-)

**Comment by xylo04** [comment]
@AsciiWolf Sure, I can help out with other ham flatpaks. It might take some time to work through the closed ones, but we can start chipping away at those. My email address is on my profile if you want to DM me.

**Comment by AsciiWolf** [comment]
@xylo04 Cool, thanks!

@barthalion @bilelmoussaoui Can you please add @xylo04 push rights to the https://github.com/flathub/com.w1hkj.fldigi and https://github.com/flathub/com.w1hkj.flrig repos? :-)

**Comment by hfiguiere** [comment]
bot, build radio.k0swe.Kel_Agent

**Comment by nedrichards** [comment]
> @barthalion @bilelmoussaoui Can you please add @xylo04 push rights to the https://github.com/flathub/com.w1hkj.fldigi and https://github.com/flathub/com.w1hkj.flrig repos? :-)

Done.

**Comment by xylo04** [comment]
@hfiguiere Thanks for your help getting this into shape. Do you need further action from me?

**Comment by barthalion** [comment]
I'm not actually sure how to test it without hardware but I guess it's fine.

/merge

**Comment by xylo04** [comment]
> I'm not actually sure how to test it without hardware but I guess it's fine.

If you're interested in testing, you can do it with software only, though it won't be very interesting. You'll need [WSJT-X](https://physics.princeton.edu/pulsar/k1jt/wsjtx.html) (configure it for no rig control) and [Forester](https://forester.radio/).

**Pull request closed without merging**